# DATA UNDERSTANDING

## 1. Import Dataset CSV

In [1]:
import numpy as np
import pandas as pd

In [2]:
df = pd.read_csv("sample_superstore.csv", encoding="latin-1")
df_ori = df.copy() 
df.head()

,Row ID,Order ID,Order Date,Ship Date,Ship Mode,Customer ID,Customer Name,Segment,Country,City,...,Postal Code,Region,Product ID,Category,Sub-Category,Product Name,Sales,Quantity,Discount,Profit
0,1,CA-2016-152156,11/8/2016,11/11/2016,Second Class,CG-12520,Claire Gute,Consumer,United States,Henderson,...,42420,South,FUR-BO-10001798,Furniture,Bookcases,Bush Somerset Collection Bookcase,261.9600,2,0.00,41.9136
1,2,CA-2016-152156,11/8/2016,11/11/2016,Second Class,CG-12520,Claire Gute,Consumer,United States,Henderson,...,42420,South,FUR-CH-10000454,Furniture,Chairs,"Hon Deluxe Fabric Upholstered Stacking Chairs,...",731.9400,3,0.00,219.5820
2,3,CA-2016-138688,6/12/2016,6/16/2016,Second Class,DV-13045,Darrin Van Huff,Corporate,United States,Los Angeles,...,90036,West,OFF-LA-10000240,Office Supplies,Labels,Self-Adhesive Address Labels for Typewriters b...,14.6200,2,0.00,6.8714
3,4,US-2015-108966,10/11/2015,10/18/2015,Standard Class,SO-20335,Sean O'Donnell,Consumer,United States,Fort Lauderdale,...,33311,South,FUR-TA-10000577,Furniture,Tables,Bretford CR4500 Series Slim Rectangular Table,957.5775,5,0.45,-383.0310
4,5,US-2015-108966,10/11/2015,10/18/2015,Standard Class,SO-20335,Sean O'Donnell,Consumer,United States,Fort Lauderdale,...,33311,South,OFF-ST-10000760,Office Supplies,Storage,Eldon Fold 'N Roll Cart System,22.3680,2,0.20,2.5164


## 2. Data Exploration

##### 1. Cek info dasar untuk jumlah baris dan kolom pada data yang ditampilkan

In [3]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9994 entries, 0 to 9993
Data columns (total 21 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Row ID         9994 non-null   int64  
 1   Order ID       9994 non-null   object 
 2   Order Date     9994 non-null   object 
 3   Ship Date      9994 non-null   object 
 4   Ship Mode      9994 non-null   object 
 5   Customer ID    9994 non-null   object 
 6   Customer Name  9994 non-null   object 
 7   Segment        9994 non-null   object 
 8   Country        9994 non-null   object 
 9   City           9994 non-null   object 
 10  State          9994 non-null   object 
 11  Postal Code    9994 non-null   int64  
 12  Region         9994 non-null   object 
 13  Product ID     9994 non-null   object 
 14  Category       9994 non-null   object 
 15  Sub-Category   9994 non-null   object 
 16  Product Name   9994 non-null   object 
 17  Sales          9994 non-null   float64
 18  Quantity

In [4]:
print("Jumlah Baris :", df.shape[0])
print("Jumlah Kolom :", df.shape[1])

Jumlah Baris : 9994
Jumlah Kolom : 21


##### 2. Cek untuk nilai Data

In [5]:
df.describe()

,Row ID,Postal Code,Sales,Quantity,Discount,Profit
count,9994.000000,9994.000000,9994.000000,9994.000000,9994.000000,9994.000000
mean,4997.500000,55190.379428,229.858001,3.789574,0.156203,28.656896
std,2885.163629,32063.693350,623.245101,2.225110,0.206452,234.260108
min,1.000000,1040.000000,0.444000,1.000000,0.000000,-6599.978000
25%,2499.250000,23223.000000,17.280000,2.000000,0.000000,1.728750
50%,4997.500000,56430.500000,54.490000,3.000000,0.200000,8.666500
75%,7495.750000,90008.000000,209.940000,5.000000,0.200000,29.364000
max,9994.000000,99301.000000,22638.480000,14.000000,0.800000,8399.976000


##### 3. Cek jumlah baris yang kosong (null) di setiap kolomnya

In [6]:
df.isnull().sum()

Row ID           0
Order ID         0
Order Date       0
Ship Date        0
Ship Mode        0
Customer ID      0
Customer Name    0
Segment          0
Country          0
City             0
State            0
Postal Code      0
Region           0
Product ID       0
Category         0
Sub-Category     0
Product Name     0
Sales            0
Quantity         0
Discount         0
Profit           0
dtype: int64

- Tidak ditemukan missing value pada dataset sehingga tidak diperlukan proses cleaning data berupa pengisian maupun penghapusan data

##### 4. Cek tipe data

In [7]:
df.dtypes

Row ID             int64
Order ID          object
Order Date        object
Ship Date         object
Ship Mode         object
Customer ID       object
Customer Name     object
Segment           object
Country           object
City              object
State             object
Postal Code        int64
Region            object
Product ID        object
Category          object
Sub-Category      object
Product Name      object
Sales            float64
Quantity           int64
Discount         float64
Profit           float64
dtype: object

##### 5. Cek duplikat data

In [8]:
duplicate = df.duplicated().sum()
print(duplicate)

0


##### 6. Cek typo pada kolom yang berupa kategorikal

In [9]:
print(df['Ship Mode'].value_counts())

Ship Mode
Standard Class    5968
Second Class      1945
First Class       1538
Same Day           543
Name: count, dtype: int64


In [10]:
print(df['Segment'].value_counts())

Segment
Consumer       5191
Corporate      3020
Home Office    1783
Name: count, dtype: int64


In [11]:
print(df['Country'].value_counts())

Country
United States    9994
Name: count, dtype: int64


In [12]:
print(df['City'].value_counts())

City
New York City      915
Los Angeles        747
Philadelphia       537
San Francisco      510
Seattle            428
                  ... 
Glenview             1
Missouri City        1
Rochester Hills      1
Palatine             1
Manhattan            1
Name: count, Length: 531, dtype: int64


In [13]:
print(df['State'].value_counts())

State
California              2001
New York                1128
Texas                    985
Pennsylvania             587
Washington               506
Illinois                 492
Ohio                     469
Florida                  383
Michigan                 255
North Carolina           249
Arizona                  224
Virginia                 224
Georgia                  184
Tennessee                183
Colorado                 182
Indiana                  149
Kentucky                 139
Massachusetts            135
New Jersey               130
Oregon                   124
Wisconsin                110
Maryland                 105
Delaware                  96
Minnesota                 89
Connecticut               82
Oklahoma                  66
Missouri                  66
Alabama                   61
Arkansas                  60
Rhode Island              56
Utah                      53
Mississippi               53
Louisiana                 42
South Carolina            42
Nevada  

In [14]:
print(df['Region'].value_counts())

Region
West       3203
East       2848
Central    2323
South      1620
Name: count, dtype: int64


In [15]:
print(df['Category'].value_counts())

Category
Office Supplies    6026
Furniture          2121
Technology         1847
Name: count, dtype: int64


In [16]:
print(df['Sub-Category'].value_counts())

Sub-Category
Binders        1523
Paper          1370
Furnishings     957
Phones          889
Storage         846
Art             796
Accessories     775
Chairs          617
Appliances      466
Labels          364
Tables          319
Envelopes       254
Bookcases       228
Fasteners       217
Supplies        190
Machines        115
Copiers          68
Name: count, dtype: int64


In [17]:
print(df['Discount'].value_counts())

Discount
0.00    4798
0.20    3657
0.70     418
0.80     300
0.30     227
0.40     206
0.60     138
0.10      94
0.50      66
0.15      52
0.32      27
0.45      11
Name: count, dtype: int64


##### 6. Cek Outlier Data

In [18]:
# Menghitung total seluruh baris data sebagai pembagi persentase
total_data = df.shape[0]

print("=== DETAIL BATAS & PERSENTASE OUTLIER ===\n")

# ==========================================
# 1. ANALISIS OUTLIER KOLOM: SALES
# ==========================================
Q1_sales = df['Sales'].quantile(0.25)
Q3_sales = df['Sales'].quantile(0.75)
IQR_sales = Q3_sales - Q1_sales
lower_sales = Q1_sales - (1.5 * IQR_sales)
upper_sales = Q3_sales + (1.5 * IQR_sales)

# Menghitung jumlah baris yang di luar batas
outlier_sales = df[(df['Sales'] < lower_sales) | (df['Sales'] > upper_sales)].shape[0]
# Rumus persentase: (jumlah outlier / total data) * 100
pct_sales = (outlier_sales / total_data) * 100

print(f"--- Kolom Sales ---")
print(f"Batas Bawah: {lower_sales:.2f} | Batas Atas: {upper_sales:.2f}")
print(f"Jumlah Outlier: {outlier_sales} baris")
print(f"Persentase Outlier: {pct_sales:.2f}%\n")


# ==========================================
# 2. ANALISIS OUTLIER KOLOM: QUANTITY
# ==========================================
Q1_qty = df['Quantity'].quantile(0.25)
Q3_qty = df['Quantity'].quantile(0.75)
IQR_qty = Q3_qty - Q1_qty
lower_qty = Q1_qty - (1.5 * IQR_qty)
upper_qty = Q3_qty + (1.5 * IQR_qty)

outlier_qty = df[(df['Quantity'] < lower_qty) | (df['Quantity'] > upper_qty)].shape[0]
pct_qty = (outlier_qty / total_data) * 100

print(f"--- Kolom Quantity ---")
print(f"Batas Bawah: {lower_qty:.2f} | Batas Atas: {upper_qty:.2f}")
print(f"Jumlah Outlier: {outlier_qty} baris")
print(f"Persentase Outlier: {pct_qty:.2f}%\n")


# ==========================================
# 3. ANALISIS OUTLIER KOLOM: DISCOUNT
# ==========================================
Q1_disc = df['Discount'].quantile(0.25)
Q3_disc = df['Discount'].quantile(0.75)
IQR_disc = Q3_disc - Q1_disc
lower_disc = Q1_disc - (1.5 * IQR_disc)
upper_disc = Q3_disc + (1.5 * IQR_disc)

outlier_disc = df[(df['Discount'] < lower_disc) | (df['Discount'] > upper_disc)].shape[0]
pct_disc = (outlier_disc / total_data) * 100

print(f"--- Kolom Discount ---")
print(f"Batas Bawah: {lower_disc:.2f} | Batas Atas: {upper_disc:.2f}")
print(f"Jumlah Outlier: {outlier_disc} baris")
print(f"Persentase Outlier: {pct_disc:.2f}%\n")


# ==========================================
# 4. ANALISIS OUTLIER KOLOM: PROFIT
# ==========================================
Q1_profit = df['Profit'].quantile(0.25)
Q3_profit = df['Profit'].quantile(0.75)
IQR_profit = Q3_profit - Q1_profit
lower_profit = Q1_profit - (1.5 * IQR_profit)
upper_profit = Q3_profit + (1.5 * IQR_profit)

outlier_profit = df[(df['Profit'] < lower_profit) | (df['Profit'] > upper_profit)].shape[0]
pct_profit = (outlier_profit / total_data) * 100

print(f"--- Kolom Profit ---")
print(f"Batas Bawah: {lower_profit:.2f} | Batas Atas: {upper_profit:.2f}")
print(f"Jumlah Outlier: {outlier_profit} baris")
print(f"Persentase Outlier: {pct_profit:.2f}%\n")

=== DETAIL BATAS & PERSENTASE OUTLIER ===

--- Kolom Sales ---
Batas Bawah: -271.71 | Batas Atas: 498.93
Jumlah Outlier: 1167 baris
Persentase Outlier: 11.68%

--- Kolom Quantity ---
Batas Bawah: -2.50 | Batas Atas: 9.50
Jumlah Outlier: 170 baris
Persentase Outlier: 1.70%

--- Kolom Discount ---
Batas Bawah: -0.30 | Batas Atas: 0.50
Jumlah Outlier: 856 baris
Persentase Outlier: 8.57%

--- Kolom Profit ---
Batas Bawah: -39.72 | Batas Atas: 70.82
Jumlah Outlier: 1881 baris
Persentase Outlier: 18.82%



##### 7. Cek nilai negatif

In [19]:
df[df['Sales']<0].head()

,Row ID,Order ID,Order Date,Ship Date,Ship Mode,Customer ID,Customer Name,Segment,Country,City,...,Postal Code,Region,Product ID,Category,Sub-Category,Product Name,Sales,Quantity,Discount,Profit


In [20]:
# KOLOM SALES
negative_sales = (df['Sales'] < 0).sum()

percentage = (negative_sales / len(df)) * 100

print(f"Jumlah sales negatif: {negative_sales}")
print(f"Persentase: {percentage:.2f}%")

Jumlah sales negatif: 0
Persentase: 0.00%


In [21]:
df[df['Quantity']<0].head()

,Row ID,Order ID,Order Date,Ship Date,Ship Mode,Customer ID,Customer Name,Segment,Country,City,...,Postal Code,Region,Product ID,Category,Sub-Category,Product Name,Sales,Quantity,Discount,Profit


In [22]:
# KOLOM QUANTITY
negative_quantity = (df['Quantity'] < 0).sum()

percentage = (negative_quantity / len(df)) * 100

print(f"Jumlah quantity negatif: {negative_quantity}")
print(f"Persentase: {percentage:.2f}%")

Jumlah quantity negatif: 0
Persentase: 0.00%


In [23]:
df[df['Discount']<0].head()

,Row ID,Order ID,Order Date,Ship Date,Ship Mode,Customer ID,Customer Name,Segment,Country,City,...,Postal Code,Region,Product ID,Category,Sub-Category,Product Name,Sales,Quantity,Discount,Profit


In [24]:
# KOLOM DISCOUNT
negative_discount = (df['Discount'] < 0).sum()

percentage = (negative_discount / len(df)) * 100

print(f"Jumlah discount negatif: {negative_discount}")
print(f"Persentase: {percentage:.2f}%")

Jumlah discount negatif: 0
Persentase: 0.00%


In [25]:
df[df['Profit']<0].head()

,Row ID,Order ID,Order Date,Ship Date,Ship Mode,Customer ID,Customer Name,Segment,Country,City,...,Postal Code,Region,Product ID,Category,Sub-Category,Product Name,Sales,Quantity,Discount,Profit
3,4,US-2015-108966,10/11/2015,10/18/2015,Standard Class,SO-20335,Sean O'Donnell,Consumer,United States,Fort Lauderdale,...,33311,South,FUR-TA-10000577,Furniture,Tables,Bretford CR4500 Series Slim Rectangular Table,957.5775,5,0.45,-383.0310
14,15,US-2015-118983,11/22/2015,11/26/2015,Standard Class,HP-14815,Harold Pawlan,Home Office,United States,Fort Worth,...,76106,Central,OFF-AP-10002311,Office Supplies,Appliances,Holmes Replacement Filter for HEPA Air Cleaner...,68.8100,5,0.80,-123.8580
15,16,US-2015-118983,11/22/2015,11/26/2015,Standard Class,HP-14815,Harold Pawlan,Home Office,United States,Fort Worth,...,76106,Central,OFF-BI-10000756,Office Supplies,Binders,Storex DuraTech Recycled Plastic Frosted Binders,2.5440,3,0.80,-3.8160
23,24,US-2017-156909,7/16/2017,7/18/2017,Second Class,SF-20065,Sandra Flanagan,Consumer,United States,Philadelphia,...,19140,East,FUR-CH-10002774,Furniture,Chairs,"Global Deluxe Stacking Chair, Gray",71.3720,2,0.30,-1.0196
27,28,US-2015-150630,9/17/2015,9/21/2015,Standard Class,TB-21520,Tracy Blumstein,Consumer,United States,Philadelphia,...,19140,East,FUR-BO-10004834,Furniture,Bookcases,"Riverside Palais Royal Lawyers Bookcase, Royal...",3083.4300,7,0.50,-1665.0522


In [26]:
negative_profit = (df['Profit'] < 0).sum()

percentage = (negative_profit / len(df)) * 100

print(f"Jumlah profit negatif: {negative_profit}")
print(f"Persentase: {percentage:.2f}%")

Jumlah profit negatif: 1871
Persentase: 18.72%


INTERPRETASI :
- Data sales, quantity, dan discount tidak memiliki nilai data yang negatif sedangkan profit memiliki nilai negatif
- Jumlah data yang memiliki nilai profit negatif yaitu 1.871 data dengan persentasenya 18.72%
- Data ini dipertahankan karena kondisi rugi dalam profit adalah kondisi bisnis nyata
- Sehingga outlier tidak perlu di handling karena tidak ada yang bernilai negatif dari 3 kolom tersebut dan untuk profit itu berkemungkinan nilainya bisa negatif atau minus

## 3. Data Cleaning

##### 1. Mengubah type data order date dan ship date 

In [27]:
df['Order Date'] = pd.to_datetime(df['Order Date'])
df['Ship Date'] = pd.to_datetime(df['Ship Date'])

In [28]:
df.dtypes

Row ID                    int64
Order ID                 object
Order Date       datetime64[ns]
Ship Date        datetime64[ns]
Ship Mode                object
Customer ID              object
Customer Name            object
Segment                  object
Country                  object
City                     object
State                    object
Postal Code               int64
Region                   object
Product ID               object
Category                 object
Sub-Category             object
Product Name             object
Sales                   float64
Quantity                  int64
Discount                float64
Profit                  float64
dtype: object

- Data order date dan ship date yang awalnya type datanya berupa object diubah menjadi datetime karena tanggal harusnya type datanya adalah datetime

In [29]:
df['Profit'].describe

<bound method NDFrame.describe of 0        41.9136
1       219.5820
2         6.8714
3      -383.0310
4         2.5164
          ...   
9989      4.1028
9990     15.6332
9991     19.3932
9992     13.3200
9993     72.9480
Name: Profit, Length: 9994, dtype: float64>

## Data Manipulation

### 1. Membuat kolom baru melalui feature engineering 

##### a. Menambahkan kolom profit ratio

In [30]:
df['Profit Ratio'] = df['Profit'] / df['Sales']

##### b. Menambahkan kolom profit status

In [31]:
def profit_status(profit):
    if profit > 0:
        return 'Profit'
    elif profit == 0:
        return 'BEP'
    else:
        return 'Loss'

df['Profit Status'] = df['Profit'].apply(profit_status)

##### c. Menambahkan kolom Discount Category

In [32]:
def discount_category(x):
    if x == 0:
        return 'No Discount'
    elif x <= 0.2:
        return 'Low Discount'
    elif x <= 0.4:
        return 'Medium Discount'
    else:
        return 'High Discount'

df['Discount Category'] = df['Discount'].apply(discount_category)

In [33]:
print(df['Discount Category'].value_counts())

Discount Category
No Discount        4798
Low Discount       3803
High Discount       933
Medium Discount     460
Name: count, dtype: int64


##### d. Menambahkan kolom tahun

In [34]:
df['Year'] = df['Order Date'].dt.year

##### e. Menambahkan kolom bulan

In [35]:
df['Month'] = df['Order Date'].dt.month_name()

INTERPRETASI :
- Untuk analisis selanjutnya diperlukan penambahan kolom untuk digunakan dalam analisis insight nantinya
- Untuk penambahan kolom yaitu kolom profit ratio, profit status, discount category, year, dan month

## Analisis Awal (grouping, sorting, filtering)

##### 1. Sales per category

In [36]:
df.groupby('Category')[['Sales', 'Profit']].sum()

,Sales,Profit
Category,,
Furniture,741999.7953,18451.2728
Office Supplies,719047.0320,122490.8008
Technology,836154.0330,145454.9481


##### 2. Profit per Region

In [37]:
df.groupby('Region')['Profit'].sum()

Region
Central     39706.3625
East        91522.7800
South       46749.4303
West       108418.4489
Name: Profit, dtype: float64

##### 3. Discount per Category

In [38]:
df.groupby('Category')['Discount'].mean()

Category
Furniture          0.173923
Office Supplies    0.157285
Technology         0.132323
Name: Discount, dtype: float64

##### 4. Top 10 Product profit

In [39]:
df.groupby('Product Name')['Profit'].sum().sort_values(ascending=False).head(10)

Product Name
Canon imageCLASS 2200 Advanced Copier                                          25199.9280
Fellowes PB500 Electric Punch Plastic Comb Binding Machine with Manual Bind     7753.0390
Hewlett Packard LaserJet 3310 Copier                                            6983.8836
Canon PC1060 Personal Laser Copier                                              4570.9347
HP Designjet T520 Inkjet Large Format Printer - 24" Color                       4094.9766
Ativa V4110MDD Micro-Cut Shredder                                               3772.9461
3D Systems Cube Printer, 2nd Generation, Magenta                                3717.9714
Plantronics Savi W720 Multi-Device Wireless Headset System                      3696.2820
Ibico EPK-21 Electric Binding System                                            3345.2823
Zebra ZM400 Thermal Label Printer                                               3343.5360
Name: Profit, dtype: float64

##### 5. Memfilter data transaksi yang rugi

In [40]:
loss_transactions = df[df['Profit']<0]
loss_transactions.head()

,Row ID,Order ID,Order Date,Ship Date,Ship Mode,Customer ID,Customer Name,Segment,Country,City,...,Product Name,Sales,Quantity,Discount,Profit,Profit Ratio,Profit Status,Discount Category,Year,Month
3,4,US-2015-108966,2015-10-11,2015-10-18,Standard Class,SO-20335,Sean O'Donnell,Consumer,United States,Fort Lauderdale,...,Bretford CR4500 Series Slim Rectangular Table,957.5775,5,0.45,-383.0310,-0.400000,Loss,High Discount,2015,October
14,15,US-2015-118983,2015-11-22,2015-11-26,Standard Class,HP-14815,Harold Pawlan,Home Office,United States,Fort Worth,...,Holmes Replacement Filter for HEPA Air Cleaner...,68.8100,5,0.80,-123.8580,-1.800000,Loss,High Discount,2015,November
15,16,US-2015-118983,2015-11-22,2015-11-26,Standard Class,HP-14815,Harold Pawlan,Home Office,United States,Fort Worth,...,Storex DuraTech Recycled Plastic Frosted Binders,2.5440,3,0.80,-3.8160,-1.500000,Loss,High Discount,2015,November
23,24,US-2017-156909,2017-07-16,2017-07-18,Second Class,SF-20065,Sandra Flanagan,Consumer,United States,Philadelphia,...,"Global Deluxe Stacking Chair, Gray",71.3720,2,0.30,-1.0196,-0.014286,Loss,Medium Discount,2017,July
27,28,US-2015-150630,2015-09-17,2015-09-21,Standard Class,TB-21520,Tracy Blumstein,Consumer,United States,Philadelphia,...,"Riverside Palais Royal Lawyers Bookcase, Royal...",3083.4300,7,0.50,-1665.0522,-0.540000,Loss,High Discount,2015,September


##### 6. Memfilter data untuk melihat diskon tertinggi

In [41]:
high_discount = df[df['Discount Category'] == 'High Discount']

high_discount.head()

,Row ID,Order ID,Order Date,Ship Date,Ship Mode,Customer ID,Customer Name,Segment,Country,City,...,Product Name,Sales,Quantity,Discount,Profit,Profit Ratio,Profit Status,Discount Category,Year,Month
3,4,US-2015-108966,2015-10-11,2015-10-18,Standard Class,SO-20335,Sean O'Donnell,Consumer,United States,Fort Lauderdale,...,Bretford CR4500 Series Slim Rectangular Table,957.5775,5,0.45,-383.0310,-0.400000,Loss,High Discount,2015,October
14,15,US-2015-118983,2015-11-22,2015-11-26,Standard Class,HP-14815,Harold Pawlan,Home Office,United States,Fort Worth,...,Holmes Replacement Filter for HEPA Air Cleaner...,68.8100,5,0.80,-123.8580,-1.800000,Loss,High Discount,2015,November
15,16,US-2015-118983,2015-11-22,2015-11-26,Standard Class,HP-14815,Harold Pawlan,Home Office,United States,Fort Worth,...,Storex DuraTech Recycled Plastic Frosted Binders,2.5440,3,0.80,-3.8160,-1.500000,Loss,High Discount,2015,November
27,28,US-2015-150630,2015-09-17,2015-09-21,Standard Class,TB-21520,Tracy Blumstein,Consumer,United States,Philadelphia,...,"Riverside Palais Royal Lawyers Bookcase, Royal...",3083.4300,7,0.50,-1665.0522,-0.540000,Loss,High Discount,2015,September
28,29,US-2015-150630,2015-09-17,2015-09-21,Standard Class,TB-21520,Tracy Blumstein,Consumer,United States,Philadelphia,...,Avery Recycled Flexi-View Covers for Binding S...,9.6180,2,0.70,-7.0532,-0.733333,Loss,High Discount,2015,September


INTERPRETASI :
- Terdapat 933 data yang termasuk kedalam "High Discount" berdasarkan filtering pada kolom kategori diskon

##### 7. Korelasi discount dan profit

In [42]:
df[['Discount','Profit']].corr()

,Discount,Profit
Discount,1.000000,-0.219487
Profit,-0.219487,1.000000


INTERPRETASI :
- Dari hasil korelasi antara discount dan profit dapat diketahui bahwa nilai korelasi negatif
- Semakin tinggi diskon yang diberikan maka profit cenderung menurun
- Dan ketika profit meningkat atau tinggi maka diskon cenderung menurun

##### 8. Korelasi sales dan profit

In [43]:
df[['Sales','Profit']].corr()

,Sales,Profit
Sales,1.000000,0.479064
Profit,0.479064,1.000000


INTERPRETASI :
- Dari hasil korelasi antara sales dan profit diketahui bahwa keduanya berbanding lurus
- Ketika sales/harga penjualan meningkat maka profit yang diperoleh perusahaan juga meningkat
- Dan ketika harga penjualan menurun maka profit yang diperoleh perusahaan juga akan menurun

##### 9. Korelasi quantity dan profit

In [44]:
df[['Quantity','Profit']].corr()

,Quantity,Profit
Quantity,1.000000,0.066253
Profit,0.066253,1.000000


INTERPRETASI :
- Dari hasil antara jumlah/quantity terhadap profit memiliki hubungan yang berbanding lurus / positif
- Ketika jumlah barang yang dijual meningkat maka profit yang diperoleh perusahaan juga akan meningkat
- Dan ketika jumlah barang yang dijual lebih sedikit maka perusahaan akan memperoleh profit lebih sedikit atau juga menurun

## Cek Dataset Terakhir

In [45]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9994 entries, 0 to 9993
Data columns (total 26 columns):
 #   Column             Non-Null Count  Dtype         
---  ------             --------------  -----         
 0   Row ID             9994 non-null   int64         
 1   Order ID           9994 non-null   object        
 2   Order Date         9994 non-null   datetime64[ns]
 3   Ship Date          9994 non-null   datetime64[ns]
 4   Ship Mode          9994 non-null   object        
 5   Customer ID        9994 non-null   object        
 6   Customer Name      9994 non-null   object        
 7   Segment            9994 non-null   object        
 8   Country            9994 non-null   object        
 9   City               9994 non-null   object        
 10  State              9994 non-null   object        
 11  Postal Code        9994 non-null   int64         
 12  Region             9994 non-null   object        
 13  Product ID         9994 non-null   object        
 14  Category

In [46]:
df.head()

,Row ID,Order ID,Order Date,Ship Date,Ship Mode,Customer ID,Customer Name,Segment,Country,City,...,Product Name,Sales,Quantity,Discount,Profit,Profit Ratio,Profit Status,Discount Category,Year,Month
0,1,CA-2016-152156,2016-11-08,2016-11-11,Second Class,CG-12520,Claire Gute,Consumer,United States,Henderson,...,Bush Somerset Collection Bookcase,261.9600,2,0.00,41.9136,0.1600,Profit,No Discount,2016,November
1,2,CA-2016-152156,2016-11-08,2016-11-11,Second Class,CG-12520,Claire Gute,Consumer,United States,Henderson,...,"Hon Deluxe Fabric Upholstered Stacking Chairs,...",731.9400,3,0.00,219.5820,0.3000,Profit,No Discount,2016,November
2,3,CA-2016-138688,2016-06-12,2016-06-16,Second Class,DV-13045,Darrin Van Huff,Corporate,United States,Los Angeles,...,Self-Adhesive Address Labels for Typewriters b...,14.6200,2,0.00,6.8714,0.4700,Profit,No Discount,2016,June
3,4,US-2015-108966,2015-10-11,2015-10-18,Standard Class,SO-20335,Sean O'Donnell,Consumer,United States,Fort Lauderdale,...,Bretford CR4500 Series Slim Rectangular Table,957.5775,5,0.45,-383.0310,-0.4000,Loss,High Discount,2015,October
4,5,US-2015-108966,2015-10-11,2015-10-18,Standard Class,SO-20335,Sean O'Donnell,Consumer,United States,Fort Lauderdale,...,Eldon Fold 'N Roll Cart System,22.3680,2,0.20,2.5164,0.1125,Profit,Low Discount,2015,October


- Setelah dicek menggunakan df.info dan df.head terbukti sudah terdapat penambahan kolom baru pada data tersebut
- Dataset awal terdapat 21 kolom dan setelah ditambahkan beberapa kolom baru untuk dataset sekarang memiliki jumlah 26 kolom

## Simpan Data Bersih

##### 1. Menyimpan dataset kedalam file excel

In [47]:
df.to_excel('superstore_clean.xlsx', index=False)

##### 2. Mencari lokasi file yang sudah dibentuk

In [48]:
import os

print(os.getcwd())

C:\Users\ASUS


- Hasil TRUE menunjukkan bahwa file berhasil dibuat

In [49]:
import os

os.listdir()

['.anaconda',
 '.conda',
 '.continuum',
 '.ipynb_checkpoints',
 '.ipython',
 '.jupyter',
 '.matplotlib',
 '.ms-ad',
 '.spss',
 '3D Objects',
 'anaconda3',
 'anaconda_projects',
 'AppData',
 'Application Data',
 'Assignment 2.ipynb',
 'ASSIGNMENT PYTHON 3 11 MEI 2026-BISMILLAH FIX-Copy1.ipynb',
 'ASSIGNMENT PYTHON 3 11 MEI 2026-Copy2.ipynb',
 'ASSIGNMENT PYTHON 3 11 MEI 2026-Copy3.ipynb',
 'ASSIGNMENT PYTHON 3 11 MEI 2026-FIX-Copy1.ipynb',
 'ASSIGNMENT PYTHON 3 11 MEI 2026.ipynb',
 'Belajar Phyton.ipynb',
 'Contacts',
 'Cookies',
 'COONECTSQL 19 MEI 2026.ipynb',
 'Data Kaggle 29 April 2026.ipynb',
 'Data Visualisasi 13 Mei 2026.ipynb',
 'Desktop',
 'Documents',
 'Downloads',
 'Employe Performance dataset_cleaned.csv',
 'employees.csv',
 'Employe_Performance_dataset.csv',
 'Employe_Performance_dataset_Cleaned Encoding.csv',
 'Favorites',
 'IntelGraphicsProfiles',
 'Latihan 1 Data Structures with Python.ipynb',
 'Latihan 1.ipynb',
 'LATIHAN 18 SOAL 7 MEI 2026.ipynb',
 'Latihan 2 10 soal 2

In [50]:
import os

print(os.path.abspath('superstore_clean.xlsx'))

C:\Users\ASUS\superstore_clean.xlsx


## MELANJUTKAN VISUALISASI DI POWER BI